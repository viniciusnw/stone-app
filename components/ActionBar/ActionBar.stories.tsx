import React from "react";
import { View, Text } from "react-native";
import styled from "styled-components/native";
import { ActionBar, ActionBarWrapperDouble, ActionBarWrapperSingle, Button, PriceResume } from "@Components";

const Wrapper = styled(View)`
  height: 100%;
  justify-content: flex-end;
`;

const ActionBarDefault = () => (
  <Wrapper>
    <ActionBar>
      <ActionBarWrapperSingle>
        <Button color="black">
          <Text>Avan√ßar</Text>
        </Button>
      </ActionBarWrapperSingle>
    </ActionBar>
  </Wrapper>
);

const ActionBarPrice = () => (
  <Wrapper>
    <ActionBar>
      <ActionBarWrapperDouble>
        <PriceResume label="3x de R$99,99" description="ou R$299,90" />
        <Button color="black">
          <Text>Comprar</Text>
        </Button>
      </ActionBarWrapperDouble>
    </ActionBar>
  </Wrapper>
);

export const Default = {
  args: {
    withPrice: false,
  },
};

export const BarPrice = {
  args: {
    withPrice: true,
  },
};

export default {
  title: "ActionBar",
  component: ({ withPrice }: any) =>
    withPrice ? <ActionBarPrice /> : <ActionBarDefault/>
};