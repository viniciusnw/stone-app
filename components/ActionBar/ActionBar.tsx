import React from "react";
import styled from "styled-components/native";
import { View } from "react-native";

const StyledContainer = styled(View)`
  background-color: #ddd;
  padding: 0 20px;
  flex-direction: row;
  align-items: center;
  height: ${({ theme }) => theme.height.navbar};
`;

type Props = {
  children: string | JSX.Element | JSX.Element[]
}

export const ActionBar: React.FC<Props> = ({ children }) => {
  return <StyledContainer>{children}</StyledContainer>;
};
