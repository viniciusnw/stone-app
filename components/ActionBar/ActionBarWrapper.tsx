import React from 'react';
import styled from 'styled-components/native';
import { View } from 'react-native';

export const ActionBarWrapperSingle = styled(View)`
  flex-direction: row;
  justify-content: flex-end;
  width: 100%;
`;

export const ActionBarWrapperDouble = styled(View)`
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
`;