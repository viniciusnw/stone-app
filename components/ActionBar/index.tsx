export * from './ActionBar';
export * from './ActionBarWrapper';
