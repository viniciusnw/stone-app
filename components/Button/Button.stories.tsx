import React from "react";
import { Text } from "react-native";
import { Button, StorybookWrapper } from "@Components";

const ButtonDefault = ({ type }: any) => (
  <StorybookWrapper>
    <Button color={type} onPress={() => true}>
      <Text>Default Button</Text>
    </Button>
  </StorybookWrapper>
);

export const Default = {
  args: {
    type: "default",
  },
};

export const Primary = {
  args: {
    type: "primary",
  },
};

export const Black = {
  args: {
    type: "black",
  },
};

export const Gray = {
  args: {
    type: "gray",
  },
};

export default {
  title: "Button",
  argTypes: {
    type: {
      options: ["default", "primary", "black", "gray"],
      control: { type: "radio" },
    },
  },
  component: ({ type }: any) => <ButtonDefault type={type} />,
};
