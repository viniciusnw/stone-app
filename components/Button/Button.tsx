import React from "react";
import styled from "styled-components/native";
import { Text, TouchableOpacity } from "react-native";

const StyledButton = styled(TouchableOpacity)<{ color: any, rounded: boolean }>`
  background-color: ${({ theme, color, rounded }) =>
    theme.color[!color ? "default" : color]};
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px 15px;
  border-radius: ${({ rounded }) => (rounded ? "50%" : 0)};
`;

const StyledText = styled(Text)`
  color: ${({ theme }) => theme.color.white};
  font-weight: 700;
  font-size: 14px;
`;

const Button = ({ color, children, ...props }: any) => {
  return (
    <StyledButton color={color} {...props}>
      <StyledText>{children}</StyledText>
    </StyledButton>
  );
};

export default Button;
