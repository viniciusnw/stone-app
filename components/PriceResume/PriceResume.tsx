import React from 'react';
import styled from 'styled-components/native';
import { View, Text } from 'react-native';

const StyledContainer = styled(View)``;

const StyledLabel = styled(Text)`
  color: ${({ theme }) => theme.color.black};
  font-weight: 700;
  font-size: 16px;
`;

const StyledDescription = styled(Text)`
  color: ${({ theme }) => theme.color.gray};
  margin-top: 2px;
`;

const PriceResume = ({ label, description, ...props }: any) => {
  return !label && !description ? null : (
    <StyledContainer {...props}>
      {label && <StyledLabel>{label}</StyledLabel>}
      {description && <StyledDescription>{description}</StyledDescription>}
    </StyledContainer>
  );
};

export default PriceResume;
