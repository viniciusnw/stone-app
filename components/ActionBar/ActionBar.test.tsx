import React from "react";
import renderer from "react-test-renderer";
import { ActionBar } from "./ActionBar";
import {
  ActionBarWrapperDouble,
  ActionBarWrapperSingle,
} from "./ActionBarWrapper";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Button from "@Components/Button";
import { Text } from "react-native";
import PriceResume from "@Components/PriceResume";

test("ActionBarWrapperSingle", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <ActionBar>
          <ActionBarWrapperSingle>
            <Button color="black">
              <Text>Avan√ßar</Text>
            </Button>
          </ActionBarWrapperSingle>
        </ActionBar>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("ActionBarWrapperDouble", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <ActionBar>
          <ActionBarWrapperDouble>
            <PriceResume label="3x de R$99,99" description="ou R$299,90" />
            <Button color="black">
              <Text>Comprar</Text>
            </Button>
          </ActionBarWrapperDouble>
        </ActionBar>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
