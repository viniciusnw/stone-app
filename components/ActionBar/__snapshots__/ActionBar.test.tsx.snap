// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBarWrapperDouble 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "flexDirection": "row",
      "height": 62,
      "paddingBottom": 0,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={{}}
    >
      <Text
        style={
          {
            "color": "#111111",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        3x de R$99,99
      </Text>
      <Text
        style={
          {
            "color": "#666666",
            "marginTop": 2,
          }
        }
      >
        ou R$299,90
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#111111",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 0,
          "borderTopLeftRadius": 0,
          "borderTopRightRadius": 0,
          "display": "flex",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "700",
          }
        }
      >
        <Text>
          Comprar
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`ActionBarWrapperSingle 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "flexDirection": "row",
      "height": 62,
      "paddingBottom": 0,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#111111",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 0,
          "borderTopLeftRadius": 0,
          "borderTopRightRadius": 0,
          "display": "flex",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 14,
            "fontWeight": "700",
          }
        }
      >
        <Text>
          Avan√ßar
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
